### PR TITLE
feat(components): [el-table] `maxHeight` support more units

### LINKED ISSUE #5850: [Feature Request] table设置最大高度时不能使用vh
### Existing Component
Yes

### Component Name
table

### Description
table组件设置max-height属性时使用vh为单位无法滚动，必须用px才正常，是否可以适配下

### LINKED ISSUE #5888: [Feature Request] 希望el-table max-hieght 支持动态计算
### Existing Component
Yes

### Component Name
el-table

### Description
el-table max-hieght 目前只有px生效，动态计算例如：calc(100% - 40px) 不生效，可否让它支持动态计算？

### PR DESCRIPTION
**Pull request by msidolphin**
re #5888 #5850

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by emojiiii — APPROVED** [review]

File: packages/components/table/src/table/style-helper.ts [hunk 1]
```diff
@@ -223,15 +223,28 @@ function useStyle<T>(
...
 
+  function calcMaxHeight(
+    maxHeight: string | number,
+    footerHeight: number,
+    headerHeight: number
+  ) {
+    const parsedMaxHeight = parseHeight(maxHeight)
+    const tableHeaderHeight = props.showHeader ? headerHeight : 0
+    if (parsedMaxHeight === null) return
+    if (typeof parsedMaxHeight === 'string') {
```
Comment (JeremyWuuuuu) [inline packages/components/table/src/table/style-helper.ts:234]: Use `isString` from `@element-plus/utils`

File: packages/components/table/src/table/style-helper.ts [hunk 2]
```diff
@@ -245,12 +258,16 @@ function useStyle<T>(
...
+        footerHeight,
+        headerHeight
+      )
+      if (maxHeight !== null) {
         return {
-          'max-height': `${
-            maxHeight - footerHeight - (props.showHeader ? headerHeight : 0)
-          }px`,
+          'max-height': `${maxHeight}${
+            typeof maxHeight === 'number' ? 'px' : ''
```
Comment (JeremyWuuuuu) [inline packages/components/table/src/table/style-helper.ts:269]: is `isNumber` from `@element-plus/utils`.

**Review by jw-foss — COMMENTED** [review]
Make sure that you rebase the `dev` branch before updating the code.

**Review by jw-foss — APPROVED** [review]

**Comment by Fate-ui** [comment]
在2.02版本中，当把max-height设为vh时，表格的下边框消失

**Comment by Fate-ui** [comment]
> 在2.02版本中，当把max-height设为vh时，表格的下边框消失

当设置为60vh时，表格下边框消失，当设置为65vh时，表格下边框出现。我不确定是不是我代码的问题

**Pull request merged**